plugins/lua/zone.lua

```diff
@@ -0,0 +1,542 @@
+local _ENV = mkmodule('plugins.zone')
+local gui = require('gui')
+local overlay = require('plugins.overlay')
+local widgets = require('gui.widgets')
+
+ZoneOverlay = defclass(ZoneOverlay, overlay.OverlayWidget)
+ZoneOverlay.ATTRS{
+    default_pos={x=46,y=39},
+    default_enabled=true,
+    viewscreens='dwarfmode',
+    frame={w=42, h=15},
+    frame_style=gui.MEDIUM_FRAME,
+    frame_background=gui.CLEAR_PEN,
+}
+
+local function zone_assign_is_active()
+    return dfhack.gui.matchFocusString('dwarfmode/UnitSelector') and (df.global.game.main_interface.unit_selector.context == 0 or df.global.game.main_interface.unit_selector.context == 1)
+end
+
+local function canShear(unit)
+    return #df.global.world.raws.creatures.all[unit.race].caste[unit.caste].shearable_tissue_layer > 0
+end
+
+local function copyTable(table_to_copy)
```

Comment:
I actually had some problems using `copyall` because it maintains the zero index, which gets missed in ipairs when it's an actual table. Like if I sub in `copyall` in place of `copyTable` right now things would break lol

Reply:
ah, ok, understood. maybe the function should be called `toLuaList()` then?